compiler-core/src/build/module_loader.rs

```diff
@@ -48,14 +47,13 @@ where
     /// Whether the module has changed or not is determined by comparing the
     /// modification time of the source file with the value recorded in the
     /// `.timestamp` file in the artefact directory.
-    pub fn load(&self, path: Utf8PathBuf) -> Result<Input> {
-        let name = module_name(self.source_directory, &path);
-        let artefact = name.replace("/", "@");
-        let source_mtime = self.io.modification_time(&path)?;
+    pub fn load(&self, file: GleamFile) -> Result<Input> {
+        let name = file.module_name.clone();
+        let source_mtime = self.io.modification_time(&file.path)?;
 
-        let read_source = |name| self.read_source(path, name, source_mtime);
+        let read_source = |name| self.read_source(file.path.clone(), name, source_mtime);
```

Comment:
I didn't spend too much time trying to get rid of these two clones, but there might be something we could still optimize. I thought I'd get overall feedback first before going too detailed.